server/tests/src/main/scala/sttp/tapir/server/tests/ServerWebSocketTests.scala

```diff
@@ -29,106 +30,143 @@ abstract class ServerWebSocketTests[F[_], S <: Streams[S], ROUTE](
   private def stringWs = webSocketBody[String, CodecFormat.TextPlain, String, CodecFormat.TextPlain].apply(streams)
   private def stringEcho = functionToPipe((s: String) => s"echo: $s")
 
-  def tests(): List[Test] = List(
-    testServer(
-      endpoint.out(stringWs),
-      "string client-terminated echo"
-    )((_: Unit) => pureResult(stringEcho.asRight[Unit])) { (backend, baseUri) =>
-      basicRequest
-        .response(asWebSocket { (ws: WebSocket[IO]) =>
-          for {
-            _ <- ws.sendText("test1")
-            _ <- ws.sendText("test2")
-            m1 <- ws.receiveText()
-            m2 <- ws.receiveText()
-          } yield List(m1, m2)
-        })
-        .get(baseUri.scheme("ws"))
-        .send(backend)
-        .map(_.body shouldBe Right(List("echo: test1", "echo: test2")))
-    }, {
+  def tests(): List[Test] = {
+    val basicTests = List(
+      testServer(
+        endpoint.out(stringWs),
+        "string client-terminated echo"
+      )((_: Unit) => pureResult(stringEcho.asRight[Unit])) { (backend, baseUri) =>
+        basicRequest
+          .response(asWebSocket { (ws: WebSocket[IO]) =>
+            for {
+              _ <- ws.sendText("test1")
+              m1 <- ws.receiveText()
```

Comment:
why the changed ordering of operations? didn't the original test work?

Reply:
Yep, the order of received messages was not deterministic for zio-server

Reply:
hm well that's bad. Either a bug in zio-http, or somewhere else. I'd revert those tests to their original ordering and look for the root cause. That's quite normal WS usage, that you send multiple messages in sequence and then get a response

Reply:
Yes, it's probably zio-http, I'll keep that in mind and see what will came up in the issue I reported - that may be related I think